### PR TITLE
MHV Account Transition: Informational/consent page

### LINKED ISSUE #34945: Create the information page for Login.gov (Eligible/Ineligible)
## Issue Description
This will be a dedicated page solely for My HealtheVet (MHV) users that **are eligible** to transition to Login.gov seamlessly.  The eligibility standards are that they need to have Identity Proofed via the [outlined requirements](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/identity/Design%20Documents/MHV%20Account%20Transition.md).

---
## Tasks
- [x] Conditionally render content based on eligibility
  - [x] **Eligible** Update CTA based on Figma wireframes & allow transition process with other information
  - [x] **Ineligible** Update CTA based on Figma wireframes & add signups to Login.gov & ID.me
- [x] Add Identity provider wizard
- [x] Ensure buttons navigate user to proper place
  - [x] Transition (verify with BE team to see if we need another route for this?)
  - [x] Dismiss (should allow a user to go back to where they were originally *maybe History API?)
  - [x] Login.gov/ID.me buttons (should use the `signup`, verify w/ BE on what should happen if they are already logged in)
- [x] Add the breadcrumbs to allow users to escape early
- [x] Grab data User model (`v0/user`) to ensure eligibility

## Acceptance Criteria
- [ ] As an (eligible) MHV user, I can read more information and start the Login.gov transition process
- [ ] As an (ineligible) MHV user, I can read more information about creating a new account (ID.me/Login.gov), use a wizard to find out which trusted partner to use

---
## How to configure this issue
- [ ] **Attached to a Milestone** (when will this be completed?)
- [ ] **Attached to an Epic** (what body of work is this a part of?)
- [ ] **Labeled with Team** (`product support`, `analytics-insights`, `operations`, `service-design`, `tools-be`, `tools-fe`)
- [ ] **Labeled with Practice Area** (`backend`, `frontend`, `devops`, `design`, `research`, `product`, `ia`, `qa`, `analytics`, `contact center`, `research`, `accessibility`, `content`)
- [ ] **Labeled with Type** (`bug`, `request`, `discovery`, `documentation`, etc.)

### PR DESCRIPTION
**Pull request by asg5704**
## Description
This PR adds a dedicated page for MHV users to allow them to transition and get more information about transition.  The new route/page will live within the SignInApp `/sign-in/transition`.  This PR additionally adds the IdentityWizard (possibly could become deprecated) as well as adds unit testing to newly created components and utilities.

## Original issue(s)
Closes department-of-veterans-affairs/va.gov-team#34945

## Testing done
Unit testing

#### **Eligible to transition**
Screen Shot 2022-03-02 at 1 06 40 PM

#### **Not eligible to transition**
Screen Shot 2022-03-02 at 1 05 31 PM

## Acceptance criteria
- [x] Conditionally renders CTA content based on eligibility (BE implementation TBD at a later time)
- [x] Updates answers (state) based on Identity Wizard
- [x] Allows an eligible user to continue to transition on Login.gov (verify w/ BE to see if new session-controller route is needed)

## Definition of done
- [x] Events are logged appropriately
- [x] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/login/components/TransitionAccountPage.jsx [hunk 1]
```diff
@@ -0,0 +1,75 @@
+import React from 'react';
+import { useSelector } from 'react-redux';
+import { transitionMHVAccount } from 'platform/user/authentication/selectors';
+import TransitionAccountCTA from './TransitionAccountCTA';
+import IdentityWizard from './IdentityWizard';
+
+export default function TransitionAccount() {
+  // TODO: wait for BE team to add to User model
+  const canTransition = useSelector(transitionMHVAccount);
```
Comment (Kwoly) [inline src/applications/login/components/TransitionAccountPage.jsx:9]: Once both of our PRs are in, and the backend has done the implementation, we should sync and make sure we utilize the same logic for this determination on both the modal and the page.

File: src/platform/user/authentication/components/CreateAccountLink.jsx [hunk 2]
```diff
@@ -1,19 +1,40 @@
...
+  const { href, children, eventBase } = {
+    href:
+      type !== 'create'
+        ? authUtilities.sessionTypeUrl({ type: csp })
+        : authUtilities.signupUrl(csp),
+    children:
+      type !== 'create'
+        ? `Sign in with ${CSP_CONTENT[csp].COPY} account`
+        : `Create an account with ${CSP_CONTENT[csp].COPY}`,
+    eventBase: type !== 'create' ? AUTH_EVENTS.LOGIN : AUTH_EVENTS.REGISTER,
```
Comment (Kwoly) [inline src/platform/user/authentication/components/CreateAccountLink.jsx:21]: If this component can now be used for login and signup we should change the name of the component to be more generic

File: src/platform/user/authentication/selectors.js [hunk 3]
```diff
@@ -33,3 +33,6 @@ export const isAuthenticatedWithSSOe = state =>
 
 export const ssoeTransactionId = state =>
   selectProfile(state)?.session?.transactionid;
+
+export const transitionMHVAccount = state =>
+  state.user.profile.mhvTransitionEligible ?? false;
```
Comment (Kwoly) [inline src/platform/user/authentication/selectors.js:38]: should we add optional chaining here or will the `?? false` handle the short circuit if we run into a `cannot read x of undefined`?

Comment (AGarcia-Clarity) [reply src/platform/user/authentication/selectors.js:38]: The nullish coalescing will work

File: src/platform/user/authentication/components/CreateAccountLink.jsx (hunk 2 above)
Comment (AGarcia-Clarity) [reply src/platform/user/authentication/components/CreateAccountLink.jsx]: Resolved.  Created a `TYPE_LINKS` constant

**Review by Kwoly — APPROVED** [review]

**Pull request merged**